lib/internal/streams/destroy.js

```diff
@@ -46,13 +45,11 @@ function destroy(err, cb) {
       }
     }
 
-    if (cb) {
-      // Invoke callback before scheduling emitClose so that callback
-      // can schedule before.
+    if (typeof cb === 'function') {
       cb(err);
-      // Don't emit 'error' if passed a callback.
-      process.nextTick(emitCloseNT, this);
```

Comment:
We always we need to emit 'error'. Before the callback would call `destroy(er)` in a roundabout way.

Reply:
This `cb` route should be a purely internal route through an internal undocumented api